### DISCUSSION TITLE
GitHub 404 for Repository that was Previously Working

### DISCUSSION THREAD

**Discussion by ekristen** [renovatebot/renovate#14678] [Request Help]
I'm using the Hosted App.

Sometime on March 14th, the Renovate Logs from RepoA started throwing GitHub 404 errors for a dependency defined as RepoB. The encrypted GitHub token works for other repos in the same org by the same user that RepoB is in, but RepoB says 404.

```
DEBUG: GitHub 404
{
  "url": "https://api.github.com/repositories/REPO_B_000000/releases?per_page=100&page=2"
}
DEBUG: GitHub 404
{
  "url": "https://api.github.com/repos/my-org/repo-b/releases?per_page=100"
}
DEBUG: Datasource 404
{
  "datasource": "github-releases",
  "packageName": "my-org/repo-b",
  "url": "https://api.github.com/repositories/REPO_B_000000/releases?per_page=100&page=2"
}
```

These run ids from the hosted app should give you the necessary logs and the actual repository names and ids.

  - 603585951
  - 603524855

**Comment by rarkins** [comment]
If you're getting that far then the decrypting of the token has worked (logs confirm this). 

You can see this:

```
Adding token authentication for https://api.github.com/repos/<your-org>/ to hostRules
```

First of all, there's some risk to this because this rule/token might take priority over the App's own token so you could see PRs come from your token's identity and not the app's. I haven't verified this but it's definitely a risk - you'd be better limiting it to a particular repo if you can.

Next, we see something I don't recall seeing before:

```
{"level":20,"url":"https://api.github.com/repositories/<repo-id>/releases?per_page=100&page=2","msg":"GitHub 404","time":"2022-03-15T20:26:16.892Z"}
{"level":20,"url":"https://api.github.com/repos/<your-org>/<your-repo>/releases?per_page=100","msg":"GitHub 404","time":"2022-03-15T20:26:16.892Z"}
```

Presumably the first message triggers the second. And you can see that the host rule you added would *not* match against that path. I have no idea what's causing the repo to be looked up by ID though.

**Reply by ekristen** [reply]
All the repos in the org are private, that's why the host rule is set that way. I couldn't get the auth to work any other way.

It's worked that way for well over a year. 

However it suddenly started working again last night. 🤷 

Perhaps the GitHub api was returning bad data or something was cached and it expired?

**Reply by ekristen** [reply]
@rarkins it happened again -- seems to be sporadic, either there's a bug in renovate or a bug in the GitHub API data, not sure.

`604513087` received a 404, but 30 minutes later `604603291` worked just fine.

**Reply by rarkins** [reply]
I could never be sure, but it seems like a GitHub API problem. Renovate's logic/code is not changing between those runs. Can you compare the log of URLs which is dumped at the end of each run?

**Reply by ekristen** [reply]
@rarkins it's still happening, it would be really interesting to see if you could review all logs and see if this is happening for other people? I wonder if GitHub is testing out changes to the API or something? It's very sporadic, but it happens daily.

**Reply by rarkins** [reply]
Which message do you recommend looking for? Eg "Datasource 404"?By default we don't pipe debug messages into cloudwatch logs so can't search them unless I add some code to the backend to remap them to info level in advance.

**Reply by ekristen** [reply]
I was thinking any calls to /repositories/ endpoint, it seems that is suppose to be for only public repos based on the documentation. Everything I can see any org related repository or private should always be /org/repo-name or /user/repo-name --- I feel like this has to be a bug with GitHub. I don't have enough data to raise an issue though.

**Reply by rarkins** [reply]
Maybe we can add this to the OSS code itself, if it's something we don't expect to ever happen. I.e. if we get a 404 and it starts with https://api.github.com/repositories/ then we log something
